agent/taskresource/cgroup/cgroup.go

```diff
@@ -49,8 +51,8 @@ type CgroupResource struct {
 	resourceSpec        specs.LinuxResources
 	ioutil              ioutilwrapper.IOUtil
 	createdAt           time.Time
-	desiredStatusUnsafe CgroupStatus
-	knownStatusUnsafe   CgroupStatus
+	desiredStatusUnsafe taskresource.ResourceStatus
+	knownStatusUnsafe   taskresource.ResourceStatus
```

Comment:
please let me know if i've missed something. why are we naming this *Unsafe? its private field and also the accessor methods have locks around them. same question for both fields added here.

Reply:
actually just noticed the naming was already there. should we change it?

Reply:
They are named unsafe because they need to be accessed via get* methods that handle locks.